### PR TITLE
What is Software Development? Content, activities, instructor guides

### PR DESCRIPTION
**Pull request by tildeee**
This is "day 2" stuff, so this content is about getting folks to really think about the possibilities of software.

### CODE REVIEW

File: what-is-software-development/study-like-a-programmer.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+Application | Can I create examples of ...? Can I repeat and modify examples of ...? If someone says that this concept is relevant, can you find a way to apply this concept and fit it into the bigger picture? | Take time to create your own working examples, and check them. Make diagrams, skits, songs, etc. and share them.
+Analysis | On a bigger scale of systems... What factors impact this? What consequences are there? What are the technical trade-offs? | Research relevant examples. Compare and contrast similar examples. Make tables and lists to compare.
+Evaluate | On a bigger scale of systems... When should I use, defend, or critique this? | Research. Hypothesize, then make short experiments. Read various sources.
+Create | On a bigger scale of systems... How do I orchestrate this with a system of other concepts? Can I create something that uses this? | Research. Planning. Hypothesizing, then make short experiements. Read various sources.
```
Comment (audreyandoy) [inline what-is-software-development/study-like-a-programmer.md:179]:
```suggestion
Level of Understanding | What kinds of Questions You Have | Action
--- | --- | ---
Knowledge | What does ... mean? What is the standard syntax for ...? Where can I find the definition for ...? | Use flashcards for specific vocabulary, concepts, and syntax
Understanding | What are some examples about ...? When is ... relevant and not relevant? What are variations of this syntax? | [Create a crib sheet/cheat sheet](https://en.wikipedia.org/wiki/Cheat_sheet). Try explaining the concept to someone else
Application | Can I create examples of ...? Can I repeat and modify examples of ...? If someone says that this concept is relevant, can you find a way to apply this concept and fit it into the bigger picture? | Take time to create your own working examples, and check them. Make diagrams, skits, songs, etc. and share them.
Analysis | On a bigger scale of systems... What factors impact this? What consequences are there? What are the technical trade-offs? | Research relevant examples. Compare and contrast similar examples. Make tables and lists to compare.
Evaluate | On a bigger scale of systems... When should I use, defend, or critique this? | Research. Hypothesize, then make short experiments. Read various sources.
Create | On a bigger scale of systems... How do I orchestrate this with a system of other concepts? Can I create something that uses this? | Research. Planning. Hypothesizing, then make short experiements. Read various sources.
Destruct | What would make ____ break? Would using _____ break another part of the code? | Break your code. Read about errors and what makes them appear.
```

I LOVE this chart. I added a 'Destruct' row as I think it's valuable to teach students to learn how to 'break' things in order to be more comfortable with errors.

File: what-is-software-development/study-like-a-programmer.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+- Feeling like you lose focus on the task at hand? Consider: Using physical paper and sticky notes to write down the specific task you are trying to accomplish
+- Feeling like you lose focus over time? Consider: Using a time management method like [Pomodoro Technique](https://en.wikipedia.org/wiki/Pomodoro_Technique) to help enable frequent breaks and frequent periods of flow
+- Finding that your knowledge and understanding level of vocabulary is lacking? Consider: Utilize flashcards for specific vocabulary, concepts, and syntax
+- Finding that you are craving more review in your own words? Consider: Making crib sheets (1-page review notes of the crucial knowledge) at the end of the week
+- Feeling isolated? Please consider having proactive communication with other students and staff! Staff is on your side for your success, and your proactive communication with staff can lead to opportunities and resources
+- Want a video series we recommend? Consider: ["Crash Course" on study skills](https://www.youtube.com/playlist?list=PL8dPuuaLjXtNcAJRf3bE1IJU6nMfHj86W)
```
Comment (audreyandoy) [inline what-is-software-development/study-like-a-programmer.md:326]:
```suggestion
<div style="width=100px;">Are you feeling like... </div>| Consider |
---- | ---- |
...classroom time is overwhelming? | Look at the calendar and the lessons the night before, and pre-read the lectures
...reading material on programming is unnatural? | Read this resource on [how to read programming books](http://www.pathsensitive.com/2018/01/the-benjamin-franklin-method-of-reading.html)
...you lose focus on the task at hand? | Using physical paper and sticky notes to write down the specific task you are trying to accomplish
...you lose focus over time? | Using a time management method like [Pomodoro Technique](https://en.wikipedia.org/wiki/Pomodoro_Technique) to help enable frequent breaks and frequent periods of flow
...your knowledge and understanding level of vocabulary is lacking? | Utilize flashcards for specific vocabulary, concepts, and syntax
...you are craving more review in your own words? | Making crib sheets (1-page review notes of the crucial knowledge) at the end of the week
... you're isolated? | Please consider having proactive communication with other students and staff! Staff is on your side for your success, and your proactive communication with staff can lead to opportunities and resources

Want a video series we recommend? Consider: ["Crash Course" on study skills](https://www.youtube.com/playlist?list=PL8dPuuaLjXtNcAJRf3bE1IJU6nMfHj86W)

```

I put this into a table to make it a little easier to read. Learn preview wasn't working for me so I'm not quite sure if the inline style worked, but it's there if you'd like to keep it.

File: what-is-software-development/what-is-software.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+Computers don't speak in human; they speak in machine code! Computers ultimately operate on 1s and 0s. Our source code gets translated several times until it becomes 1s and 0s.
+
+Source code is the kind of code that we'll write in this curriculum!
+
+### !callout-info
+
+## What does "Open Source" mean?
+
+Open Source Software (OSS) means software that was intentionally built with an open development process -- the development and its source code is available for people to look at, and to use! (within its licenses). OSS encourages collaboration, lower barriers to entry, and accountability.
```
Comment (audreyandoy) [inline what-is-software-development/what-is-software.md:49]:
```suggestion
Open Source Software (OSS) means software that was intentionally built with an open development process -- the development and its source code is available for the public to look at, and to use! (within its licenses). OSS encourages collaboration, lower barriers to entry, and accountability.
```
"public" seems more descriptive than 'people' for the open-availability of OSS.

File: what-is-software-development/what-is-software.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+##### !end-options
+
+##### !answer
+
+* the source code is available to look at and use
+
+##### !end-answer
+
+### !end-challenge
+<!-- prettier-ignore-end -->
```
Comment (audreyandoy) [inline what-is-software-development/what-is-software.md:190]: This was great. I really like the questions spread throughout the lesson. I didn't see learning goals for this lesson, was that something we wanted to add?

**Review by audreyandoy — APPROVED** [review]
Added questions to lessons. Second reviewer feel free to add input (especially in the activity sections :) )

File: what-is-software-development/study-like-a-programmer.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+- Use these studying tools and techniques to improve question-asking skills
+- Use these studying tools and techniques to improve studying skills
+
+## Consider the Role
+
+There are so many ways to define the role of a software developer and programmer. We're sure you all spent a lot of time considering what it means to you.
+
+Some words that might describe what the role of a software developer is:
+
+- A career
```
Comment (ace-n) [inline what-is-software-development/study-like-a-programmer.md:42]: "Someone who writes stories for machines. These stories almost always build off others - like fan-fictions or sequels."? 😛

(This is definitely true at larger companies/codebases - where you're often trying to stay "canonical" with the rest of the codebase.)

File: what-is-software-development/study-like-a-programmer.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+1 | I know what I know | I know what I've been taught
+2 | I don't know what I don't know | I'm learning enough to know there's more to learn
+3 | I know what I don't know | I've learned enough to know what I've learned vs. what more to learn
+4 | I don't know what I know | I don't even realize how much I've learned
+
+Levels 1 and 2 should describe where everyone is right now with programming. In fact, we should spend most of our times as learners in level 1 and 2.
+
+Level 3 is the destination that we want all of our careers to point to.
+
+Level 4 describes an experienced programmer. It is _not_ our destination.
```
Comment (ace-n) [inline what-is-software-development/study-like-a-programmer.md:162]: I'd add:
> (though you'll probably experience this at some point in your post-Ada career).

File: what-is-software-development/study-like-a-programmer.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+
+We recommend starting to review material by starting with...
+
+- Review questions that you came up with or heard over the week
+- Review questions from roundtable guides
+- Review summaries at the end of lessons
+- Review vocabulary
+- Creating a new list of questions for yourself
+
+Set up a cadence for yourself to review material. This is where your present-self should look out for your future self: reviewing material is much easier, funner, and more effective if you take notes in the beginning.
```
Comment (ace-n) [inline what-is-software-development/study-like-a-programmer.md:251]: Grammar nit: `more fun` (or `more enjoyable`)

File: what-is-software-development/what-is-software.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+## You are the Local Machine
+
+What does "local machine" mean? The phrase "local machine" will be used to mean your own very computer that you're writing code on. "Local" refers to a computer close to you, the user. We say "machine" to not assume that the users of code are laptops or desktop computers!
+
+### !end-callout
+
+## Check for Understanding
+
+<!-- Question 1 -->
```
Comment (ace-n) [inline what-is-software-development/what-is-software.md:126]: @tildeee these questions LGTM - did you want more before closing #36?

(I can add some more, if you'd like me to.)

**Review by ace-n — APPROVED** [review]

File: what-is-software-development/activity-intention-setting-our-careers.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: 2ac36021-2acf-4b30-a5ed-0aac231c2c3f
+* title: Intention Setting
+
+##### !question
+
+Write down a mobile app, web app, or software you like and why. (Ex: Zombies, Run! app, netflix.com, Microsoft OneNote)
```
Comment (ace-n) [inline what-is-software-development/activity-intention-setting-our-careers.md:29]: Nit: `piece of` software

File: what-is-software-development/study-like-a-programmer.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+
+#### We Need Little Breaks to Stay Focused
+
+Feeling like you lose focus over time? Our brains have evolved to focus on things for relatively short periods of time. We need little breaks to stay focused.
+
+Consider using our favorite time management technique, [Pomodoro Technique](https://en.wikipedia.org/wiki/Pomodoro_Technique), to help enable frequent breaks and frequent periods of flow!
+
+Encourage yourself and others to take breaks often.
+
+#### Sleep
```
Comment (ace-n) [inline what-is-software-development/study-like-a-programmer.md:280]: I might add something like this:
> In fact, taking care of yourself as a software engineer usually leads to *higher* overall productivity than overworking!

File: what-is-software-development/study-like-a-programmer.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+## Learning Goals
+
+- Interpret the role of "software developer" as a creative, human, artistic role
+- Explain that studying as a software developer requires self-care
+- Use these studying tools and techniques to improve question-asking skills
+- Use these studying tools and techniques to improve studying skills
+
+## Consider the Role
+
+There are so many ways to define the role of a software developer and programmer. We're sure you all spent a lot of time considering what it means to you.
```
[the remark this answers is not in the document]
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:38]: This is a GREAT awesome thing, and I'd love to address it in a different place (maybe the literal "what is software dev" file, maybe somewhere else)

File: what-is-software-development/study-like-a-programmer.md (hunk 1 above)
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:179]: This is awesome. I'm gonna think on this; contextually, this is supposed to borrow from a teaching/learning framework. I wonder if there's space for this in the "how to break down a problem" resource?

File: what-is-software-development/what-is-software.md (hunk 4 above)
Comment (tildeee) [reply what-is-software-development/what-is-software.md:190]: Goood point 😆  Thanks, I'll add them after merge

File: what-is-software-development/study-like-a-programmer.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
+# How to Study like a Programmer
+
+<iframe src="https://adaacademy.hosted.panopto.com/Panopto/Pages/Embed.aspx?pid=c777056e-412b-40f6-999b-ac2501267610&autoplay=false&offerviewer=true&showtitle=true&showbrand=false&start=0&interactivity=all" height="405" width="720" style="border: 1px solid #464646;" allowfullscreen allow="autoplay"></iframe>
+
```
[the remark this answers is not in the document]
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:4]: ^_____^ tanks

File: what-is-software-development/what-is-software.md (hunk 8 above)
Comment (tildeee) [reply what-is-software-development/what-is-software.md:126]: Sounds like we're good!! Thanks :D !!

File: what-is-software-development/activity-intention-setting-our-careers.md (hunk 9 above)
Comment (tildeee) [reply what-is-software-development/activity-intention-setting-our-careers.md:29]:
```suggestion
Write down a mobile app, web app, or piece of software you like and why. (Ex: Zombies, Run! app, netflix.com, Microsoft OneNote)
```

File: what-is-software-development/study-like-a-programmer.md [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+There are so many ways to define the role of a software developer and programmer. We're sure you all spent a lot of time considering what it means to you.
+
+Some words that might describe what the role of a software developer is:
+
+- A career
+- A problem-solver who uses programming to build programs and software
+- Someone who can transform digital data
+- A team-player
+- A logician
+- A chef who remixes different "recipes" of code
```
Comment (tildeee) [inline what-is-software-development/study-like-a-programmer.md:47]:
```suggestion
- A chef who remixes different "recipes" of code
- Someone who writes stories for machines. These stories almost always build off others - like fan-fictions or sequels
```

@ace-n Kind-of-arbitrarily taking your suggesiton and putting it here. I wish I knew how to add you as a co-author to this suggestion!

File: what-is-software-development/study-like-a-programmer.md (hunk 6 above)
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:162]: I appreciate the suggestion! I feel pretty strongly to let this statement be more "simple"/straightforward-- by the time a student even needs to consider level 4, they'll understand that... they're there. :P ? Not sure if my logic follows, but I'd like to not expound on Level 4.

File: what-is-software-development/study-like-a-programmer.md (hunk 7 above)
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:251]:
```suggestion
Set up a cadence for yourself to review material. This is where your present-self should look out for your future self: reviewing material is much easier, more fun, and more effective if you take notes in the beginning.
```

:D 👍

File: what-is-software-development/study-like-a-programmer.md [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+
+Feeling like you lose focus over time? Our brains have evolved to focus on things for relatively short periods of time. We need little breaks to stay focused.
+
+Consider using our favorite time management technique, [Pomodoro Technique](https://en.wikipedia.org/wiki/Pomodoro_Technique), to help enable frequent breaks and frequent periods of flow!
+
+Encourage yourself and others to take breaks often.
+
+#### Sleep
+
+Your brain is probably going to be much less productive at fixing a bug on the 11th hour of work today than you will be if you get good sleep and tackle the same bug first thing in the morning.  Also, don't be afraid to take naps!
```
Comment (tildeee) [inline what-is-software-development/study-like-a-programmer.md:282]:
```suggestion
Your brain is probably going to be much less productive at fixing a bug on the 11th hour of work today than you will be if you get good sleep and tackle the same bug first thing in the morning. In fact, taking care of yourself as a software engineer usually leads to higher overall productivity than overworking! Also, don't be afraid to take naps!
```

Tagging for visibility, and nothing else! @ace-n

File: what-is-software-development/study-like-a-programmer.md (hunk 6 above)
Comment (tildeee) [reply what-is-software-development/study-like-a-programmer.md:163]: I'm resolving this to help me feel like I'm closing this thread, not to shut down any conversation you might want to have. If you want, please continue on Slack! @ace-n <- tagging for visibility, tho i'm sure github has sent you emails about this lol

**Pull request merged**